### ISSUE TITLE
Идея: список телеграм чатов Вастрик.Клуба в боте

### ISSUE THREAD

**Issue by Anatolt** [vas3k/vas3k.club#562]
Бот старается стать как juick. Было бы логично добавить возможность для пользователей клуба видеть прямо в боте перечень всех чатов

**Comment by vas3k** [comment]
Хорошая идея, к ней тогда уж надо прикрутить хранение чатов в БД, а то сейчас страничка /network/ по сути голый HTML. Кто-то может заняться, это несложно!

**Comment by Anatolt** [comment]
нафига в бд-то?

Пн, 15 февр. 2021 г. в 20:12, Vasily Zubarev <notifications@github.com>:

**Comment by vas3k** [comment]
> нафига в бд-то?

А где?

**Comment by devcooch** [comment]
> нафига в бд-то?

В публичный репозиторий не выложить, бот имеет доступ к БД клуба, куда админ/модератор может руками добавлять записи, тогда бот мог бы высылать ссылки. Конечно, в коде надо убедиться, что это личное сообщение от человека, имеющего активный профиль в клубе.

**Comment by Anatolt** [comment]
да просто текстом

пн, 15 февр. 2021 г. в 23:32, Vasily Zubarev <notifications@github.com>:

**Comment by vas3k** [comment]
И как ты синхронизируешь обновления этого текста в двух местах и разных верстках? Руками каждый раз оба текста править предлагаешь и не потерять ничего по пути? Плохая идея

**Comment by Anatolt** [comment]
помоему это излишнее переусложнение. новые чаты не так часто добавляются
чтобы париться по этому поводу с базой данных

вт, 16 февр. 2021 г. в 11:11, Vasily Zubarev <notifications@github.com>:

**Comment by drugoi** [comment]
Может доставать из сформированного JSON-а и для сайта и для бота?

**Comment by vas3k** [comment]
> Может доставать из сформированного JSON-а и для сайта и для бота?

А где хранить JSON? :D

> помоему это излишнее переусложнение. новые чаты не так часто добавляются

Я редактирую эту страницу минимум раз в неделю. Это достаточно часто, иначе я бы не требовал такого переусложнения. Иногда линки на чаты истекают случайным образом и мне нужно быстро их менять.

**Comment by Anatolt** [comment]
кстати, а нельзя как-то брать эту инфу исходя из того в каких чатах состоит
бот?

вт, 16 февр. 2021 г. в 12:29, Vasily Zubarev <notifications@github.com>:

**Comment by Anatolt** [comment]
Можно завести табличку с чатами в airtable и использовать её как базу для бота. И даже наверное писать в неё либо этим же ботом, либо каким-то отдельным

**Comment by devcooch** [comment]
> Можно завести табличку с чатами в airtable и использовать её как базу для бота. И даже наверное писать в неё либо этим же ботом, либо каким-то отдельным

 Зачем табличка в стороннем сервисе, если у клуба есть своя БД??

**Comment by vas3k** [comment]
Да, я тоже перестал понимать что происходит в этом тикете :D

**Comment by Piom** [comment]
Бот крутиться постоянно? держать в рантайме где он есть в участниках?

**Comment by devcooch** [comment]
> Бот крутиться постоянно? держать в рантайме где он есть в участниках?

Стоит начать с того, что список чатов клуба и список чатов, где есть бот - это не одно и то же множество.

**Comment by Anatolt** [comment]
> > Можно завести табличку с чатами в airtable и использовать её как базу для бота. И даже наверное писать в неё либо этим же ботом, либо каким-то отдельным
> 
> Зачем табличка в стороннем сервисе, если у клуба есть своя БД??

затем что можно будет дать доступ к табличке админам чатов например. вон в индихакерах хотят ссылку поменятть - надо лезть в бд, а так поменяли бы в airtable сами и везде бы поменялось - и в боте и на сайте

**Comment by Anatolt** [comment]
да, актуально. тащить всё не обязательно, но перечень чатов есть в Notion в клубе и хотелось бы как-то управлять этим из одного места. ну или хотя бы чтоб то что на сайте клуба было и в боте
